### PR TITLE
[doc](nereids)Optimize doc and code usage

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

Improve and optimize the documentation for transparent rewriting of materialized views.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVCache.java [hunk 1]
```diff
@@ -64,11 +64,8 @@ public static MTMVCache from(MTMV mtmv, ConnectContext connectContext) {
         Plan originPlan = planner.plan(unboundMvPlan, PhysicalProperties.ANY, ExplainLevel.REWRITTEN_PLAN);
         Plan mvPlan = originPlan.accept(new DefaultPlanRewriter<Object>() {
             @Override
-            public Plan visit(Plan plan, Object context) {
-                if (plan instanceof LogicalResultSink) {
-                    return ((LogicalResultSink<Plan>) plan).child();
-                }
-                return super.visit(plan, context);
+            public Plan visitLogicalResultSink(LogicalResultSink<? extends Plan> logicalResultSink, Object context) {
+                return logicalResultSink.child().accept(this, context);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVCache.java:68]: file sink need keep order too. so u need override visitLogicalFileSink

Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVCache.java:68]: this code is aim to remove the resultSink for query rewrite by materilized view later. Remove sort under result sink is  in another pr

**Review by morrySnow — APPROVED** [review]

**Review by keanji-x — APPROVED** [review]

**Comment by seawinde** [comment]
this is replaced by anohter pr https://github.com/apache/doris-website/pull/438, so this is closed

**Pull request closed without merging**